Olwa pr (#102)
hms-networks/flexy-osisoft-connector: README.md

#### Web Configuration

#### FTP Configuration

> If To change configuration values, manually edit your ConnectorConfig.json file on the Flexy in the /usr/ directory via FTP.

Reviewer: The phrase "If To change configuration values, m....." doesn't make sense

## Getting Started

> ### Basic authentication

Reviewer 2: ```suggestion
### Basic Authentication
```

> This connector uses Basic authentication to communicate with OSIsoft and requires Basic authentication is enabled.

Reviewer 2: ```suggestion
This connector uses basic authentication (RFC7617) to communicate with OSIsoft and requires basic authentication to be enabled.
```

### Configuration

> User configuration of the application parameters is done through the ConnectorConfig.json file.  This file must be modified to match your configuration.

Reviewer 2: ```suggestion
User configuration of the application parameters is done through the ConnectorConfig.json file.  This file must be modified to match the configuration.
```

```suggestion
User configuration of the application parameters is done through the ConnectorConfig.json file.  This file must be modified to match the OSIsoft configuration.
```

Reviewer 2: This got missed @oliver-walker-hms

#### FTP Configuration

> To change configuration values, manually edit your ConnectorConfig.json file on the Flexy in the /usr/ directory via FTP.

Reviewer 2: ```suggestion
To change configuration values, manually edit the ConnectorConfig.json file on the Flexy in the /usr/ directory via FTP.
```

Are you editing via FTP or updating the file via FTP and editing with another tool? FTP is not an editor.

## Configuration File

### Example ConnectorConfig.json

> ``` JSON

Reviewer: See: https://www.markdownguide.org/extended-syntax/#syntax-highlighting

## Configuration File

### Example ConnectorConfig.json
```json
{
   "ServerConfig":{
      "URL":"192.168.0.124",
      "WebID":"s0U1IjG6kMOEW7mxyHCuX2mAUEktU0VSVkVSLVBD",
      "Credentials":"UEktU2VydmVyOk1hbmNoZXN0ZXIxMjMh",
      "ProxyURL":"long/proxy/example/omf"
   },
   "eWONConfig":{
      "CertificatePath":"/usr/Certificates"
   },
   "AppConfig":{
      "CommunicationType":"omf",
      "OSIsoftTagNamingScheme":"default",
      "LoggingLevel": 4,
      "httpTimeoutSeconds": 2
   }
}
```
### ServerConfig

The OSIsoft PI Database Web API must be installed for this connector to work.  The connector requires three items to connect to the OSIsoft PI Web API they are the URL of the server running the PI Web API, WebID of the PI Web API and a valid user name and password for the system running the PI Web API software.

#### URL
IP address or domain name of the system running the Web API (in this example assume the IP address is 192.168.0.124)

#### WebID

> This field is only needed when the CommunicationType is set to "piwebapi". WebID is the web ID of your OSIsoft PI Web API.

Reviewer 2: ```suggestion
This field is only needed when the CommunicationType is set to "piwebapi". WebID is the web ID of the OSIsoft PI Web API instance.
```

This value can be attained by pointing a browser to the OSIsoft PIWEBAPI URL (https://192.168.0.124/piwebapi/dataservers).  The WebID is one of several parameters returned.  A username and password will be required to access the page.  If a username is not required it may be cached.  Try clearing the cache to verify the username and password.  The same username and password will be used to create the credentials in the next step. 

> (If the PIWEBAPI certificates are self-signed a security warning may occur when trying to access this page. This warning should be ignored)

Reviewer 2: ```suggestion
(If the PIWEBAPI certificates are self-signed a security warning may occur when trying to access this page. This warning should be ignored.)
```

#### Proxy URL

> This optional field is only needed when using a non default endpoint of https://HOSTNAME/piwebapi. Adding the field "ProxyURL" under the server config section will change request endpoints to https://HOSTNAME/proxy-path. HOSTNAME will be retrieved from the URL field in the configuration file. 

Reviewer 2: ```suggestion
This optional field is only needed when using a non-default endpoint of https://HOSTNAME/piwebapi. Adding the field "ProxyURL" under the server config section will change request endpoints to https://HOSTNAME/proxy-path. HOSTNAME will be retrieved from the URL field in the configuration file. 
```

> Below is a partial config file with the optional proxy URL in use. Its endpoint is set to https://192.168.1.1/long/proxy/example/omf

Reviewer 2: ```suggestion
Below is a partial config file with the optional proxy URL in use. Its endpoint is set to https://192.168.1.1/long/proxy/example/omf.
```

   ```json
   {

>    "ServerConfig":{

Reviewer 2: Not part of this PR but this should be indented 1 level.

Reviewer: @oliver-walker-hms I would add a commit to fix indentation on JSON in README.md...looking at the JSON right now makes it a bit harder to follow with improper indentation on a few portions.

      "URL":"192.168.1.1",
      "ProxyURL":"long/proxy/example/omf"
      },
   }
   ```

#### Credentials
Base64 encoded user credentials for basic authentication

To generate your Base64 encoded user credentials visit https://www.base64encode.org/ and encode "username:password"

Example: If your username is 'username' and your password is 'password' you would encode "username:password" and should get "dXNlcm5hbWU6cGFzc3dvcmQ="

### eWONConfig

#### CertificatePath
Path to the directory containing your server's certificate.  For more information see the Certificates section

### AppConfig

#### CommunicationType

> This refers to if you are using OMF (OSIsoft message format) or PIWEBAPI. OMF is the recommended setting. 

Reviewer 2: ```suggestion
This refers to if you are using OMF (OSIsoft Message Format) or PIWEBAPI. OMF is the recommended setting. 
```

* TT, which stands for tag type.

If the value of "default" is given, tag naming scheme will be set to TN-SN-TT. tags would be named as (tag name)-(serial number)-(tag type).
The following are rules to make a valid tag name scheme:

* Tag name scheme must include the tag name
* Tag name options must be separated by the "-" delimiter
* There cannot be more than 3 tag name scheme options present
* There cannot be less than 1 tag name scheme option present
* There cannot be any options used other than the 4 listed values of "default", "SN", "TN", or "TT"

#### LoggingLevel
The OSIsoft connector uses the [HMS Solution Center logging library](https://github.com/hms-networks/sc-flexy-logger-lib) for application logging to the Ewon Flexy's realtime logs.

To change the logging level in the configuration file, specify the logging level under the AppConfig object.
LoggingLevel is an optional field in the configuration file and is not required by the application to properly run.

Below is a partial example of the configuration file set to log general application state information.
See [Log Level](https://github.com/hms-networks/sc-flexy-logger-lib#log-level) for more information.
   ```json

>    {
>    "AppConfig": {
>        "LoggingLevel": 4
>        }
>    }

Reviewer 2: ```suggestion
   {
       "AppConfig": {
           "LoggingLevel": 4
       }
   }
```

(If the PIWEBAPI certificates are self-signed a security warning may occur when trying to access this page. This warning should be ignored)

#### Proxy URL

> 

Reviewer: Adding back the whitespace you removed in previous commit be77d9a

(If the PIWEBAPI certificates are self-signed a security warning may occur when trying to access this page. This warning should be ignored.)

#### Proxy URL
This optional field is only needed when using a non-default endpoint of https://HOSTNAME/piwebapi. Adding the field "ProxyURL" under the server config section will change request endpoints to https://HOSTNAME/proxy-path. HOSTNAME will be retrieved from the URL field in the configuration file. 

Below is a partial config file with the optional proxy URL in use. Its endpoint is set to https://192.168.1.1/long/proxy/example/omf.
   ```json
   {
   "ServerConfig":{
      "URL":"192.168.1.1",
      "ProxyURL":"long/proxy/example/omf"

>       },

Reviewer: trailing comma should be removed. Also indentation is improper on the closing bracket

* There cannot be less than 1 tag name scheme option present
* There cannot be any options used other than the 4 listed values of "default", "SN", "TN", or "TT"

#### LoggingLevel
The OSIsoft connector uses the [HMS Solution Center logging library](https://github.com/hms-networks/sc-flexy-logger-lib) for application logging to the Ewon Flexy's realtime logs.

To change the logging level in the configuration file, specify the logging level under the AppConfig object.
LoggingLevel is an optional field in the configuration file and is not required by the application to properly run.

Below is a partial example of the configuration file set to log general application state information.
See [Log Level](https://github.com/hms-networks/sc-flexy-logger-lib#log-level) for more information.

>    ``` JSON

Reviewer: same suggestion about using ```json

   1. Replace the xxx below with your namespace.
   1. Replace the yyy below with your tenant ID.
   1. Copy these into the ServerConfig portion of the json file.
      ``` json
      "Namespace":"xxx",
      "TenantId":"yyy"
      ```
   1. Make sure that the JSON formatting is valid. Below is an example of a properly formatted configuration file.

>    ``` json

Reviewer: ```suggestion
   ```json
```

Reviewer: There are a few instances of this still remaining.
